cmd/erasure-server-pool.go

```diff
@@ -1713,8 +1784,16 @@ func (z *erasureServerPools) ListMultipartUploads(ctx context.Context, bucket, p
 		if err != nil {
 			return result, err
 		}
-		poolResult.Uploads = append(poolResult.Uploads, result.Uploads...)
+		// poolResult.Uploads = append(poolResult.Uploads, result.Uploads...)
+		uploads = append(uploads, result.Uploads...)
 	}
+
+	for _, upload := range uploads {
+		if ok := z.multipartUploadsCache.Has(upload.UploadID); ok {
+			poolResult.Uploads = append(poolResult.Uploads, upload)
+		}
+	}
```

Comment:
I don't understand how this works.

A) The `multipartUploadsCache` is local only, correct? I don't see anything distributing cached entries or any hashing to have it placed dterministically.
B) An upload will only be cached on one server in cluster. How can we filter, if the upload will only be cached on a random server?
C) What if a server was restarted?

I may be missing something, but this seems to assume that the NewMultipartUpload and ListMultipartUploads happen on the same server, which is unlikely.

Both distributing to all servers and asking all servers seems like a non-scalable approach.